runtime-id

```diff
@@ -47,9 +47,22 @@ if [[ ${portable_rid} == 1 ]]; then
 else
     case "${ID}" in
         # Remove the minor version
-        alpine|ol|rhel|rocky) rid_version=${VERSION_ID%.*} ;;
+        ol|rhel|rocky)
+            rid_version=${VERSION_ID%.*}
+            ;;
+        alpine)
+            if [[ $VERSION_ID == *"_"* ]]; then
+                # Alpine in-development versions contain underscore and no third part of the version
+                rid_version=${VERSION_ID%_*}
+            else
+                # Remove the patch version from major.minor.micro string
+                rid_version=${VERSION_ID%.*}
```

Comment:
Thanks for this. This is not always true anymore. Sometimes it is '3.17.0`, sometimes `3.17.0_rc1`, sometimes `3.17_alpha`. To cover all possibilities, we do the following to normalize all possible RIDs:
```diff
diff --git a/runtime-id b/runtime-id
index 5673bba..7369776 100755
--- a/runtime-id
+++ b/runtime-id
@@ -44,12 +44,15 @@ if [[ ${portable_rid} == 1 ]]; then
         echo "linux-${arch}"
     fi
 else
-    case "${ID}" in
-        # Remove the minor version
-        alpine|ol|rhel|rocky) rid_version=${VERSION_ID%.*} ;;
-
-        *) rid_version=${VERSION_ID} ;;
-    esac
-
-    echo "${ID}.${rid_version}-${arch}"
+    VERSION_ID_DOT="${VERSION_ID//[^.]}"
+    while [[ ${#VERSION_ID_DOT} -gt 1 ]]; do
+        VERSION_ID="${VERSION_ID%.*}"
+        VERSION_ID_DOT="${VERSION_ID//[^.]}"
+    done
+    VERSION_ID_DASH="${VERSION_ID//[^_]}"
+    while [[ ${#VERSION_ID_DASH} -ge 1 ]]; do
+        VERSION_ID="${VERSION_ID%_*}"
+        VERSION_ID_DASH="${VERSION_ID//[^_]}"
+    done
+    echo "${ID}.${VERSION_ID}-${arch}"
 fi
-- 
2.37.1
```
It'd be nice if RID normalization logics were universalized as, indeed, hardcoding DistroWatch in our code is becoming a chore.

Reply:
> hardcoding DistroWatch in our code is becoming a chore

:rofl: